auslib/admin/views/scheduled_changes.py

```diff
@@ -37,14 +38,26 @@ def get(self):
                     if k.startswith("base_"):
                         k = k.replace("base_", "")
                         base_row[k] = v
+                        if getattr(self.table, k).primary_key:
+                            base_pk[k] = v
                     scheduled_change[k] = v
 
             for signoff in self.sc_table.signoffs.select({"sc_id": row["sc_id"]}):
                 scheduled_change["signoffs"][signoff["username"]] = signoff["role"]
 
             # No point in retrieving this for completed scheduled changes...
             if not row["complete"]:
-                for rs in self.table.getPotentialRequiredSignoffs([base_row]):
+                affected_rows = []
+                # We don't need to consider the existing version of a row for
+                # inserts, because it doesn't exist yet!
+                if row["change_type"] != "insert":
+                    affected_rows.append(self.table.select(where=base_pk)[0])
+                # We don't need to consider the future version of the row when
+                # looking for required signoffs, because it won't exist when
+                # enacted.
+                if row["change_type"] != "delete":
+                    affected_rows.append(base_row)
```

Comment:
Yeah, change_type can be "insert", "update", or "delete". affected_rows here don't end up being inserted into the database or anything - they're just used for finding the required signoffs for this scheduled change. Does that make sense?